lib/pkg_editor/test/pkg_editor_test.cpp

```diff
@@ -396,11 +411,59 @@ static bool files_same(const char *f1, const char *f2) {
   return std::equal(begin1, std::istreambuf_iterator<char>(), begin2);
 }
 
+static void generate_random_file(const fs::path &name, size_t size) {
+  random_bytes_engine rbe;
+  std::ofstream file(name);
+  file.exceptions(std::ifstream::failbit | std::ifstream::badbit);
+  std::generate_n(std::ostreambuf_iterator<char>(file), size,
+                  [&] { return (char)rbe(); });
+}
+
+// Creates a folder consisting of a certain number of files with random contents
+// and random sizes
+static std::vector<fs::path> generate_tmp_folder(const fs::path &tmpdir) {
+  std::vector<fs::path> files;
+
+  std::mt19937 gen{};
+  std::uniform_int_distribution dis{0, 100000};
+
+  int system_result = 0;
+  std::string remove_command = "rm -rf " + tmpdir.string();
```

Comment:
```suggestion
  std::string remove_command = "cd " + tmpdir.string() + " && rm -rf *";
```

Reply:
Seems like this change will properly delete the tmp directory after some local testing

Reply:
The previous error's root cause is the tmp directory not properly removed (and caused some corruption on windows specifically?)

Reply:
This seems to work, though I have no idea why :smile:
If the tmp directory is corrupt, why would "cd"ing into it fix it...

Anyways, I would also add a check to see if `fs::exists(tmpdir)` is true before doing the system call to avoid any other error caused by "cd"ing into the directory.

Can you please make this change and update the PR @tylerzhao7684? Thanks!

Reply:
Seems like my approach still didn't solve the root problem
In my suggested code I didn't remove the tmp directory and it did not clean up after the unit tests finished
However, removing the tmp directory will throw and win32 exception, still trying to figure out why...

Reply:
Still not sure what is going on...
When trying to delete and recreate the tmp directory in the second call of `generate_tmp_folder` will fail (The tmp directory will not be created in the second call)
Exception is thrown in `generate_random_file` because the `tmp` directory did not get created in the second call but the program tries to put files inside the non-existent `tmp` directory.

Tried debugging this: 
During the second call of `generate_tmp_folder`, `system("ls")` was used to see if there is any remaining `tmp` directory after `rm -rf tmp`, there was no `tmp` directory
However, `system("ls")` was used again after `fs::create_directory(tmpdir)`, yet there is still no `tmp` directory
I replaced `fs::create_directory(tmpdir)` with `mkdir tmp` and there was this error message:
`A subdirectory or file tmp already exists` even though `ls` did not show any `tmp` directory or file

However there is a workaround, which is to name the `tmp` directory differently in each unit test. This avoids removing and re-creating the directory that caused this issue
Tested this workaround locally and the `tmp` directories are properly cleaned up
Will update the PR with the workaround soon...